test/Domain/LeanCode.DomainModels.MassTransitRelay.Tests/Integration/TestApp.cs

```diff
@@ -60,79 +46,101 @@ public TestApp()
         // it will get dropped prematurely and the tests would fail.
         dbConnection = new(connStr.ConnectionString);
 
-        var services = new ServiceCollection();
-        services.AddLogging(cfg => cfg.AddSerilog());
-        services.AddDbContext<TestDbContext>(cfg => cfg.UseSqlite(connStr.ConnectionString));
+        host = new HostBuilder()
+            .ConfigureWebHost(webHost =>
+            {
+                webHost
+                    .UseTestServer()
+                    .ConfigureServices(cfg =>
+                    {
+                        cfg.AddDbContext<TestDbContext>(db => db.UseSqlite(connStr.ConnectionString));
+                        cfg.AddLogging(l => l.AddSerilog());
+                        cfg.AddRouting();
+                        cfg.AddCQRS(TypesCatalog.Of<TestCommand>(), TypesCatalog.Of<TestCommandHandler>());
+                        cfg.AddMassTransitTestHarness(ConfigureMassTransit);
+                        cfg.AddAsyncEventsInterceptor();
+                        cfg.AddBusActivityMonitor();
+                        cfg.AddOptions<OutboxDeliveryServiceOptions>()
+                            .Configure(opts => opts.QueryDelay = TimeSpan.FromSeconds(1));
+                    })
+                    .Configure(app =>
+                    {
+                        app.UseRouting();
+                        app.UseEndpoints(ep =>
+                        {
+                            ep.MapRemoteCqrs(
+                                "/cqrs",
+                                cqrs =>
+                                {
+                                    cqrs.Queries = q => q.Secure();
+                                    cqrs.Commands = c =>
+                                        c.Secure().Validate().CommitTransaction<TestDbContext>().PublishEvents();
```

Comment:
Middlewares are bidirectional so in this case event publishing will be first in the return phase, then commit.

And yes, since the outbox is configured `PublishEvents` will just add them to db context. When the request scope ends the background outbox service will be pulsed to publish events to the transport.